### PR TITLE
add type for ./pwndbg/lib/

### PR DESCRIPTION
**Pull request by syheliel**
The only one not fixed is:
```
pwndbg/lib/memoize.py:34: error: "memoize" has no attribute "caches"; maybe "cache"?
```
I'm not sure if adding `caches` as a variable for `memoize` will break the original behavior

### CODE REVIEW

**Comment by syheliel** [comment]
I can't repduce the lint error on my local machine
image

File: pwndbg/lib/arch.py [hunk 1]
```diff
@@ -39,4 +36,4 @@ def signed(self, integer: int) -> int:
         return self.unpack(self.pack(integer), signed=True)  # type: ignore
 
     def unsigned(self, integer: int) -> int:
-        return self.unpack(self.pack(integer))
+        return self.unpack(self.pack(integer))  # type: ignore
```
Comment (gsingh93) [inline pwndbg/lib/arch.py]: Can you explain why we need the `type: ignore`? I like to actually put the error message we're ignoring in a comment so it's clear why we need the `type: ignore`.

Comment (syheliel) [reply pwndbg/lib/arch.py]: It is caused by a wrong type in `unpack`, already fixed.

**Comment by disconnect3d** [comment]
What is the state of this PR @syheliel? It seems that some tests are failing due to the changes made here: 
image

**Comment by syheliel** [comment]
@disconnect3d I have fixed the test. But the name of variable `e` is really meaningless in `commands/heap.py`:
```
            e = addr + 2 * size_sz
            e += allocator.tcache_entry.keys().index("key") * ptr_size
            e = pwndbg.gdblib.memory.pvoid(e)
```

**Comment by disconnect3d** [comment]
I'm not sure about the part in pwndbg/config.py - I am merging this for now as the changes are good, but the config.py additions may be not obvious for others. We are also missing many others config.py variables there.

On the other side: maybe we should switch to a more readable way of defining config variables?

**Pull request merged**